src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1089,15 +1089,27 @@ public void visitApply(JCTree.JCMethodInvocation tree) {
 
                     Symbol sym = access.sym;
                     List<Value> args = new ArrayList<>();
+                    boolean isSuper;
                     if (!sym.isStatic()) {
                         args.add(receiver);
+                        isSuper = switch (access.selected) {
```

Comment:
Note for later - `Foo.super.baz()` - maybe add a `@@@` comment

Reply:
But... bonus points for having handled `Foo.super.xyz` and `Foo.this.xyz` :-)

Reply:
Memo to self - we don't currently support reflecting over the body of a constructor.